test/test_intensify/test_parallel_hyperband.py

```diff
@@ -0,0 +1,419 @@
+
+    def test_process_results_via_sourceid(self):
+        """Makes sure source id is honored when deciding
+        which SH will consume the result/run_info
+
+        """
+        # Mock the SH so we can make sure the correct item is passed
+        for i in range(10):
+            self.PHB.intensifier_instances[i] = mock.Mock()
+            self.PHB.intensifier_instances[i].process_results.return_value = (self.config1, 0.5)
+            # make iter false so the mock object is not overwritten
+            self.PHB.intensifier_instances[i].iteration_done = False
+
+        # randomly create run_infos and push into PHB. Then we will make
+        # sure they got properly allocated
+        for i in np.random.choice(list(range(10)), 30):
+            run_info = RunInfo(
+                config=self.config1,
+                instance=0,
+                instance_specific="0",
+                cutoff=None,
+                seed=0,
+                capped=False,
+                budget=0.0,
+                source_id=i,
+            )
+
+            # make sure results aren't messed up via magic variable
+            # That is we check only the proper SH has this
+            magic = time.time()
+
+            result = RunValue(
+                cost=1,
+                time=0.5,
+                status=StatusType.SUCCESS,
+                starttime=1,
+                endtime=2,
+                additional_info=magic
+            )
+            self.PHB.process_results(
+                run_info=run_info,
+                incumbent=None,
+                run_history=self.rh,
+                time_bound=None,
+                result=result,
+                log_traj=False
+            )
+
+            # Check the call arguments of each sh instance and make sure
+            # it is the correct one
+
+            # First the expected one
+            self.assertEqual(
+                self.PHB.intensifier_instances[i].process_results.call_args[1]['run_info'], run_info)
+            self.assertEqual(
+                self.PHB.intensifier_instances[i].process_results.call_args[1]['result'], result)
+            all_other_run_infos, all_other_results = [], []
+            for j in range(len(self.PHB.intensifier_instances)):
+                # Skip the expected SH
+                if i == j:
+                    continue
+                if self.PHB.intensifier_instances[j].process_results.call_args is None:
+                    all_other_run_infos.append(None)
+                else:
+                    all_other_run_infos.append(
+                        self.PHB.intensifier_instances[j].process_results.call_args[1]['run_info'])
+                    all_other_results.append(
+                        self.PHB.intensifier_instances[j].process_results.call_args[1]['result'])
+            self.assertNotIn(run_info, all_other_run_infos)
+            self.assertNotIn(result, all_other_results)
+
+    def test_get_next_run_single_SH(self):
```

Comment:
I think there are copy paste errors here.

Reply:
Sorry I fixed this